test/integration/mesos/mesos_helpers.bash

```diff
@@ -4,38 +4,39 @@ load ../helpers
 
 MESOS_IMAGE=jimenez/mesos-dev:clang
 MESOS_MASTER_PORT=$(( ( RANDOM % 1000 )  + 10000 ))
-BASE_PORT=2375
 
 # Start mesos master and slave.
 function start_mesos() {
+	local current=${#DOCKER_CONTAINERS[@]}
 	MESOS_MASTER=$(
 	docker_host run -d --name mesos-master --net=host \
 	$MESOS_IMAGE /mesos/build/bin/mesos-master.sh --ip=127.0.0.1 --work_dir=/ --registry=in_memory --port=$MESOS_MASTER_PORT
 		)
 
 	retry 10 1 eval "docker_host ps | grep 'mesos-master'"
-
-	MESOS_SLAVE=$(
-	docker_host run --privileged -d --name mesos-slave --volumes-from node-0 -e DOCKER_HOST="${HOSTS[0]}" -v /sys/fs/cgroup:/sys/fs/cgroup --net=host \
-	$MESOS_IMAGE /mesos/build/bin/mesos-slave.sh --master=127.0.0.1:$MESOS_MASTER_PORT --containerizers=docker --hostname=127.0.0.1 --port=$(($MESOS_MASTER_PORT + 1))
-	       )
-
-	retry 10 1 eval "docker_host ps | grep 'mesos-slave'"
+	for ((i=0; i < current; i++)); do
+	    MESOS_SLAVES[$i]=$(
+		docker_host run --privileged -d --name mesos-slave-$i --volumes-from node-$i -e DOCKER_HOST="${HOSTS[$i]}" -v /sys/fs/cgroup:/sys/fs/cgroup --net=host \
+		$MESOS_IMAGE /mesos/build/bin/mesos-slave.sh --master=127.0.0.1:$MESOS_MASTER_PORT --containerizers=docker --attributes="docker_port:${PORTS[$i]}" --hostname=127.0.0.1 --port=$(($MESOS_MASTER_PORT + (1 + $i)))
```

Comment:
`--attributes=docker_port` is a custom data to apply to the slave (kind of daemon labels) the default would be 2375
`--port` is the port the master will bind on. the default would be 5050